spec/build_spec.lua

```diff
@@ -451,3 +448,434 @@ describe("LuaRocks build tests #integration", function()
    end)
 
 end)
+
+test_env.unload_luarocks()
+test_env.setup_specs()
+local fetch = require("luarocks.fetch")
+local fs = require("luarocks.fs")
+local path = require("luarocks.path")
+local build_builtin = require("luarocks.build.builtin")
+local build_command = require("luarocks.build.command")
+
+describe("LuaRocks build tests #unit", function()
+   local runner
+   
+   setup(function()
+      runner = require("luacov.runner")
+      runner.init(testing_paths.testrun_dir .. "/luacov.config")
+      runner.tick = true
+   end)
+   
+   teardown(function()
+      runner.shutdown()
+   end)
+   
+   describe("build.builtin", function()
+      describe("builtin.autodetect_external_dependencies", function()
+         it("returns false if the given build table has no external dependencies", function()
+            local build_table = {
+               type = "builtin"
+            }
+            assert.falsy(build_builtin.autodetect_external_dependencies(build_table))
+         end)
+         
+         it("returns a table of the external dependencies found in the given build table", function()
+            local build_table = {
+               type = "builtin",
+               modules = {
+                  module1 = {
+                     libraries = { "lib0", "lib1" },
+                  },
+                  module2 = {
+                     libraries = "lib2"
+                  },
+               }
+            }
+            
+            local extdeps = build_builtin.autodetect_external_dependencies(build_table)
+            assert.same(extdeps["LIB0"], { library = "lib0" })
+            assert.same(extdeps["LIB1"], { library = "lib1" })
+            assert.same(extdeps["LIB2"], { library = "lib2" })
+         end)
+         
+         it("adds proper include and library dirs to the given build table", function()
+            local build_table
+            
+            build_table = {
+               type = "builtin",
+               modules = {
+                  module1 = {
+                     libraries = "lib"
+                  }
+               }
+            }
+            build_builtin.autodetect_external_dependencies(build_table)
+            assert.same(build_table, {
+               type = "builtin",
+               modules = {
+                  module1 = {
+                     libraries = "lib",
+                     incdirs = { "$(LIB_INCDIR)" },
+                     libdirs = { "$(LIB_LIBDIR)" }
```

Comment:
minor, but: could you use something like `foo` instead of `lib` in the examples? This leads to confusing names like `LIB_LIBDIR` :)